sympy/polys/ring_series.py

```diff
@@ -1190,6 +1191,8 @@ def rs_atan(p, x, prec):
 
     atan
     """
+    if _get_constant_term(p, x) == EX(I) or _get_constant_term(p, x) == -EX(-I):
```

Comment:
The constant term should not be computed twice. One `c = _get_constant_term(p, x)` will suffice.
It should not be necessary to write `EX(I)` for `I`.
```
>>> EX(I) == I
True
```

Reply:
Okay made necessary changes.